BackToBasicsUIKit/BackToBasicsUIKit/Demo ViewControllers/Jimmy/JimmySecondViewController.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  JimmySecondViewController.swift
+//  BackToBasicsUIKit
+//
+//  Created by Jimmy Gutierrez on 5/23/19.
+//  Copyright © 2019 SwiftMiami. All rights reserved.
+//
+
+import UIKit
+
+protocol canRecieve {
+    func dataRecieved(data: String)
```

Comment:
The function signature should also include the sender, so the delegate knows who called it:

`func dataRecieved(viewController: UIViewController, data: String)`

Reply:
I know that you (Jimmy) were trying to follow the protocol pattern of a capability but I think you should scrap this idea and think of it as a Delegate specific to `JimmySecondViewController`. Then the function should be a specific need of `JimmySecondViewController` to communicate to another object. Maybe `func didEnterText(_ text: String)`.